src/components/SecondaryPanes/Breakpoints.js

```diff
@@ -100,14 +158,91 @@ class Breakpoints extends PureComponent {
     };
 
     const deleteAllBreakpoints = {
-      id: "node-menu-delete-all-breakpoint",
+      id: "node-menu-delete-all-breakpoints",
       label: removeAllBreakpointsLabel,
       accesskey: removeAllBreakpointsKey,
       disabled: false,
       click: () => removeAllBreakpoints()
     };
 
-    const items = [{ item: deleteBreakpoint }, { item: deleteAllBreakpoints }];
+    const deleteOtherBreakpoints = {
+      id: "node-menu-delete-other-breakpoints",
+      label: removeOtherBreapointsLabel,
+      accesskey: removeOtherBreapointsKey,
+      disabled: false,
+      click: () => removeBreakpoints(otherBreakpoints)
+    };
+
+    const enableBreakpoint = {
+      id: "node-menu-enable-breakpoint",
+      label: enableBreakpointLabel,
+      accesskey: enableBreakpointKey,
+      disabled: false,
+      click: () => toggleDisabledBreakpoint(breakpoint.location.line)
+    };
+
+    const enableAllBreakpoints = {
+      id: "node-menu-enable-all-breakpoints",
+      label: enableAllBreakpointsLabel,
+      accesskey: enableAllBreakpointsKey,
+      disabled: false,
+      click: () => toggleAllBreakpoints(false)
+    };
+
+    const enableOtherBreakpoints = {
+      id: "node-menu-enable-other-breakpoints",
+      label: enableOtherBreakpointsLabel,
+      accesskey: enableOtherBreakpointsKey,
+      disabled: false,
+      click: () => toggleBreakpoints(false, otherDisabledBreakpoints)
+    };
+
+    const disableBreakpoint = {
+      id: "node-menu-disable-breakpoint",
+      label: disableBreakpointLabel,
+      accesskey: disableBreakpointKey,
+      disabled: false,
+      click: () => toggleDisabledBreakpoint(breakpoint.location.line)
+    };
+
+    const disableAllBreakpoints = {
+      id: "node-menu-disable-all-breakpoints",
+      label: disableAllBreakpointsLabel,
+      accesskey: disableAllBreakpointsKey,
+      disabled: false,
+      click: () => toggleAllBreakpoints(true)
+    };
+
+    const disableOtherBreakpoints = {
+      id: "node-menu-disable-other-breakpoints",
+      label: disableOtherBreakpointsLabel,
+      accesskey: disableOtherBreakpointsKey,
+      click: () => toggleBreakpoints(true, otherEnabledBreakpoints)
+    };
+
+    const items = [
+      { item: enableBreakpoint, hidden: () => breakpoint.disabled === false },
+      { item: disableBreakpoint, hidden: () => breakpoint.disabled === true },
+      { item: deleteBreakpoint },
+      { item: deleteAllBreakpoints },
+      { item: deleteOtherBreakpoints, hidden: () => breakpoints.size === 1 },
+      {
+        item: enableAllBreakpoints,
+        hidden: () => disabledBreakpoints.size === 0
+      },
+      {
+        item: disableAllBreakpoints,
+        hidden: () => enabledBreakpoints.size === 0
+      },
+      {
+        item: enableOtherBreakpoints,
+        hidden: () => otherDisabledBreakpoints.size === 0
+      },
+      {
+        item: disableOtherBreakpoints,
+        hidden: () => otherEnabledBreakpoints.size === 0
```

Comment:
@clarkbw @bgrins is it better to use the `hidden` field or just not add it to the list? I'm guessing hidden, but not sure